tracing-jersey/src/main/java/com/palantir/tracing/jersey/TraceEnrichingFilter.java

```diff
@@ -87,6 +87,9 @@ public void filter(ContainerRequestContext requestContext, ContainerResponseCont
         if (maybeSpan.isPresent()) {
             Span span = maybeSpan.get();
             headers.putSingle(TraceHttpHeaders.TRACE_ID, span.getTraceId());
+        } else {
```

Comment:
It's pretty funky that this filter is called twice in the case of a failing streaming response... indicates to me that we don't fully understand the jersey behavior here.

ignoring our ignorance for a moment, the fix seems ok to me, but i'd like to see a comment indicating why this `else` block exists and what it does and how.

Reply:
Agreed, will do